src/Middleware/HttpLogging/src/HttpLoggingMiddleware.cs

```diff
@@ -50,88 +54,125 @@ public Task Invoke(HttpContext context)
-                AddToList(list, nameof(request.Method), request.Method);
+                logContext.AddParameter(nameof(request.Method), request.Method);
             }
 
             if (loggingFields.HasFlag(HttpLoggingFields.RequestScheme))
             {
-                AddToList(list, nameof(request.Scheme), request.Scheme);
+                logContext.AddParameter(nameof(request.Scheme), request.Scheme);
             }
 
             if (loggingFields.HasFlag(HttpLoggingFields.RequestPath))
             {
-                AddToList(list, nameof(request.PathBase), request.PathBase);
-                AddToList(list, nameof(request.Path), request.Path);
+                logContext.AddParameter(nameof(request.PathBase), request.PathBase);
+                logContext.AddParameter(nameof(request.Path), request.Path);
             }
 
             if (loggingFields.HasFlag(HttpLoggingFields.RequestQuery))
             {
-                AddToList(list, nameof(request.QueryString), request.QueryString.Value);
+                logContext.AddParameter(nameof(request.QueryString), request.QueryString.Value);
             }
 
             if (loggingFields.HasFlag(HttpLoggingFields.RequestHeaders))
             {
-                FilterHeaders(list, request.Headers, options._internalRequestHeaders);
+                FilterHeaders(logContext, request.Headers, options._internalRequestHeaders);
             }
 
-            if (loggingFields.HasFlag(HttpLoggingFields.RequestBody))
+            if (logContext.InternalParameters.Count > 0 && !options.CombineLogs)
             {
-                if (request.ContentType is null)
-                {
-                    _logger.NoMediaType("request");
-                }
-                else if (MediaTypeHelpers.TryGetEncodingForMediaType(request.ContentType,
-                    options.MediaTypeOptions.MediaTypeStates,
-                    out var encoding))
-                {
-                    var requestBodyLogLimit = options.RequestBodyLogLimit;
-                    if (loggingAttribute?.IsRequestBodyLogLimitSet is true)
-                    {
-                        requestBodyLogLimit = loggingAttribute.RequestBodyLogLimit;
-                    }
+                var httpRequestLog = new HttpLog(logContext.InternalParameters, "Request");
 
-                    originalBody = request.Body;
-                    requestBufferingStream = new RequestBufferingStream(
-                        request.Body,
-                        requestBodyLogLimit,
-                        _logger,
-                        encoding);
-                    request.Body = requestBufferingStream;
-                }
-                else
-                {
-                    _logger.UnrecognizedMediaType("request");
-                }
-            }
+                _logger.RequestLog(httpRequestLog);
 
-            var httpRequestLog = new HttpRequestLog(list);
+                logContext.InternalParameters.Clear();
```

Comment:
Based on the last commit, should this line be deleted?
https://github.com/dotnet/aspnetcore/blob/40e35f03e7410a7053d6d12570d9a7cbf78ac249/src/Middleware/HttpLogging/src/HttpLoggingInterceptorContext.cs#L141-L142

Reply:
Good catch